apollo-router/src/plugins/telemetry/config_new/instruments.rs

```diff
@@ -985,6 +987,54 @@ impl InstrumentsConfig {
                     .init(),
             ),
         );
+
+        // This new metric will also have the pipeline information.
+        // The reason that this instrument has a different name to the one above is that we wish to also implement this in router 2.0
+        instruments.insert(
+            OPEN_CONNECTIONS_METRIC.to_string(),
+            StaticInstrument::GaugeU64(
+                meter
+                    .u64_observable_gauge(OPEN_CONNECTIONS_METRIC)
+                    .with_description("Number of currently connected clients")
+                    .with_callback(move |gauge| {
+                        let connections =
+                            crate::axum_factory::connection_handle::connection_counts();
+                        for (connection, count) in connections.iter() {
+                            let mut attributes = Vec::with_capacity(5);
```

Comment:
oh, the # of attributes when listening on a port is actually 5 or 6. should we use 6 here? (`server.address`, `server.port`, `schema.id`, `config.hash`, `state` is 5, + optionally `launch.id`)
```suggestion
                            let mut attributes = Vec::with_capacity(6);
```

Reply:
I also fixed this on the 1.x branch